CMSgov/mint-app
# [NOREF] Update HHS FPKI to include new root CA

# No Ticket

## Changes and Description

Per an email from CMS - a new Certificate Authority has been created, and should be used by July 20, 2022. This PR introduces that CA.

- Introduces the `HHS-FPKI-Intermediate-CA-E1.cer` file into the `config/tls` folder.
- Copies the `config/tls/HHS-FPKI-Intermediate-CA-E1.cer` into `/usr/local/share/ca-certificates`, as this is where the `update-ca-certificates` utility looks for certificates (note, we rename it from `.cer` to `.crt`, as that's what the utility looks for in that folder.
  > Furthermore  all  certificates  with  a  .crt  extension  found below /usr/local/share/ca-certificates are also included as implicitly trusted. [[source](https://manpages.ubuntu.com/manpages/xenial/man8/update-ca-certificates.8.html)]
- Modifies the `dev` Docker layer (which is used locally) to include the same certificates as what's used in production by copying `/etc/ssl/certs/ca-certificates`, rather than only copying a specific set of certs. This makes it easier to test & validate that the certificates we see locally is what will be used in production.

## How to test this change

- Start the backend - `scripts/dev up:backend`
- Shell into the backend so you can explore the file structure & run commands `docker-compose exec -it mint /bin/bash`
- Scan for expected certificates. The serial of the old certificate is `44809a90`, and the new one is `44817282`, per the email from CMS.
```bash
# Old Certificate
awk -v cmd='openssl x509 -noout -subject -serial -startdate -enddate' '/BEGIN/{close(cmd)};{print | cmd}' < /etc/ssl/certs/ca-certificates.crt | grep -i '44809a90' -B1 -A2
```

```bash
# New Certificate
awk -v cmd='openssl x509 -noout -subject -serial -startdate -enddate' '/BEGIN/{close(cmd)};{print | cmd}' < /etc/ssl/certs/ca-certificates.crt | grep -i '44817282' -B1 -A2
```

- Expect to see proper output for each, showing that we've imported them.

## PR Author Review Checklist



## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.
